app/models/crime_application.rb

```diff
@@ -1,19 +1,11 @@
-class CrimeApplication < ApplicationStruct
-  include ApiResource
-
-  attribute :id, Types::String
-  attribute :application_reference, Types::String
-  attribute :application_type, Types::String
-  attribute :application_start_date, Types::JSON::DateTime
-  attribute :submission_date, Types::JSON::DateTime
+require 'laa_crime_schemas'
 
-  attribute :client_details, ClientDetails
-  attribute :case_details, CaseDetails
-  attribute :interests_of_justice, Types::Array.of(InterestOfJustice)
+class CrimeApplication < LaaCrimeSchemas::Structs::CrimeApplication
+  include ApiResource
 
   def applicant_name
-    client = client_details.client
-    [client.first_name, client.last_name].join ' '
+    applicant = client_details.applicant
+    [applicant.first_name, applicant.last_name].join ' '
```

Comment:
n a recent version of the schemas gem, there is a full_name method on any struct of type `Person` (so `Applicant` and `Codefendant` both have it). So you can do:

`client_details.applicant.full_name`